cfe_internal/update/update_policy.cf

```diff
@@ -300,13 +298,106 @@ bundle agent cfe_internal_update_policy_cpv
 bundle agent cfe_internal_update_cmdb
 # @brief Ensure local cache of CMDB data is up to date
 {
+  methods:
+
+    policy_server.enterprise_edition::
+      "cfe_internal_update_cmdb_data_distribution";
+
+      # Don't pull CMDB data on policy_hub self bootstrap because
+      # there will be no cf-serverd listening to serve files yet.
+    enterprise_edition.!(am_policy_hub.bootstrap_mode)::  # ENT-6840
+      "cfe_internal_update_cmdb_data_consumption";
+}
+
+bundle agent cfe_internal_update_cmdb_data_distribution
+# @brief Ensure data is ready for agents to download
+{
+  vars:
+    policy_server.enterprise_edition::
+
+      # The API response for host specific data from cmdb tells us the timestamp of the last data change
+      # We store this timestamp and use it for the next request.
+      "_cmdb_next_request_state_file"
+        string => "$(sys.statedir)/cmdb_next_request_from.dat";
+
+      # If we have the timestamp from a previous response we use it, else we start from 0
+      "_cmdb_previous_next_request_from"
+        string => readfile( $(_cmdb_next_request_state_file), inf ),
+        if => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      "_cmdb_previous_next_request_from"
+        string => "0",
+        unless => regline( "^\d+$", $(_cmdb_next_request_state_file) );
+
+      # We need a script to call that should return the API response
+      "_get_cmdb_data_bin" string => "$(sys.bindir)/get_cmdb.php";
+      "_get_cmdb_data_cmd" string => "/var/cfengine/httpd/php/bin/php $(_get_cmdb_data_bin) $(_cmdb_previous_next_request_from)";
+
+      # We call the script and we pass it the timestamp from the prior call
+      "_get_cmdb_data_response"
+        string => execresult( $(_get_cmdb_data_cmd), useshell ), # This would be execresult()
+        if => fileexists( $(_get_cmdb_data_bin) );
+
+      "_get_cmdb_data_response_d"
+        data => mergedata('$(_get_cmdb_data_response)');
+
+      # So that we can write a JSON file for each host we get the indicies of data in the response
+      "_i" slist => getindices( "_get_cmdb_data_response_d[data]");
+
+      # We need to store the timestamp from the most recent change so that we can use that as a starting point for future requests.
+      "_next_request_from"
+        string => "$(_get_cmdb_data_response_d[meta][cmdb_epoch])";
+
   files:
-      "$(sys.workdir)/data/." create => "true";
+      # "$(_get_cmdb_data_cmd)" perms => m( 700 );
+
+      # If the next request state file doesn't exist, we seed one with 0, the lowest epoch value possible. because we populate variables from this file content.
+      "$(_cmdb_next_request_state_file)"
+        create => "true",
+        handle => "cmdb_data_change_next_seed",
+        template_data => parsejson('{}'),
+        template_method => "inline_mustache",
+        edit_template_string => "0",
```

Comment:
`content => "0"` instead of templating.

Reply:
Yeah, I considered that. But, then I also need to guard this with a Macro so as to not blow up older versions.

Could do that though. This part is not intended to execute on hosts prior to 3.18.0.

Reply:
I'd prefer that.

Reply:
I did.